sdk/java/pulumi/src/main/java/com/pulumi/resources/internal/Stack.java

```diff
@@ -0,0 +1,109 @@
+package com.pulumi.resources.internal;
+
+import com.pulumi.core.Output;
+import com.pulumi.core.internal.annotations.InternalUse;
+import com.pulumi.deployment.internal.DeploymentInternal;
+import com.pulumi.resources.ComponentResource;
+import com.pulumi.resources.ComponentResourceOptions;
+import com.pulumi.resources.ResourceTransformation;
+
+import javax.annotation.ParametersAreNonnullByDefault;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import static java.util.Objects.requireNonNull;
+
+@InternalUse
+@ParametersAreNonnullByDefault
+public final class Stack extends ComponentResource {
```

Comment:
I wanted to radically simplify this class so there is no magic in the constructor, apart from the inherited magic)